### PR TITLE
doc: add examples

### PR DESCRIPTION
**Pull request by smintz**

### CODE REVIEW

File: protoconfloader.go [hunk 1]
```diff
@@ -114,10 +114,6 @@ func (c *Configuration) LoadConfig(configPath string, configName string) error {
 // It watches for file changes and agent updates using separate goroutines.
 // The method logs the successful start of watching and returns nil upon successful completion.
 func (c *Configuration) WatchConfig(ctx context.Context) error {
-	if !c.isLoaded.Load() {
```
Comment (avivl) [inline protoconfloader.go:117]: Why did you remove this lines?

**Review by avivl — COMMENTED** [review]
Why did you create and example_test? What's the purpose

**Comment by smintz** [comment]
See: https://go.dev/blog/examples
This will add runnable examples into the pkg.go.dev docs.

File: protoconfloader.go (hunk 1 above)
Comment (avivl) [reply protoconfloader.go:117]: @smintz ?

Comment (smintz) [reply protoconfloader.go:117]: Some users might only use the agent for subscription. In these cases, this statement prevents watching the config for no good reason.
